app/models/questionnaire.rb

```diff
@@ -1,14 +1,29 @@
 # frozen_string_literal: true
 
 class Questionnaire < ApplicationRecord
+  KNOWN_OPERATIONS = %i[average].freeze
+
   validates :name, presence: true, uniqueness: true
   validates :content, presence: true
   validates :key, presence: true, uniqueness: true, format: { with: /\A[a-z]+[a-z_0-9]*\Z/ }
-  serialize :content, Array
-  validate :all_content_ids_unique
-  validate :all_questions_have_types
-  validate :all_questions_have_titles
-  validate :all_questions_have_ids
+  serialize :content # Don't specify Hash type because otherwise databases with existing questionnaires won't work
+  validate :questionnaire_structure, if: -> { content.present? }
+  with_options if: :content_has_questions do
```

Comment:
wow nice

Reply:
Yep, it saves a lot of duplicated code in checks.